apps/prairielearn/src/middlewares/csrfToken.ts

```diff
@@ -7,14 +6,14 @@ import { generateSignedToken, checkSignedToken } from '@prairielearn/signed-toke
 import { config } from '../lib/config.js';
 
 export default asyncHandler(async (req, res, next) => {
-  const tokenData = {
+  const tokenData: {
+    url: string;
+    authn_user_id?: string;
+  } = {
```

Comment:
Nit, I think type inference is sufficient here? It's true that `res.locals` is basically typed as `any`, but I don't think that matters since this is pretty much just an opaque blob that's passed to `checkSignedToken`.

```suggestion
  const tokenData = {
```

Reply:
That makes sense since generateSignedToken and checkSignedToken accept any.